WebARonARKit/WebARonARKit/ViewController.m

```diff
@@ -952,40 +998,66 @@ - (void)session:(ARSession *)session didUpdateFrame:(ARFrame *)frame {
     position[1] = pModelMatrix[13];
     position[2] = pModelMatrix[14];
 
-    // TODO: Testing to see if we can pass the whole frame to JS...
-    //  size_t width = CVPixelBufferGetWidth(frame.capturedImage);
-    //  size_t height = CVPixelBufferGetHeight(frame.capturedImage);
-    //  size_t bytesPerRow = CVPixelBufferGetBytesPerRow(frame.capturedImage);
-    //  void* pixels = CVPixelBufferGetBaseAddress(frame.capturedImage);
-    //  OSType pixelFormatType =
-    //  CVPixelBufferGetPixelFormatType(frame.capturedImage);
-    //  NSLog(@"width = %d, height = %d, bytesPerRow = %d, ostype = %d", width,
-    //  height, bytesPerRow, pixelFormatType);
-
-    NSString *jsCode = [NSString
-                        stringWithFormat:
-                        @"if (window.WebARonARKitSetData) "
-                        @"window.WebARonARKitSetData({"
-                        @"\"position\":[%f,%f,%f],"
-                        @"\"orientation\":[%f,%f,%f,%f],"
-                        @"\"viewMatrix\":[%f,%f,%f,%f,%f,%f,%f,%"
-                        @"f,%f,%f,%f,%f,%f,%f,%f,%f],"
-                        @"\"projectionMatrix\":[%f,%f,%f,%f,%f,%f,%f,%"
-                        @"f,%f,%f,%f,%f,%f,%f,%f,%f]"
-                        @"});",
-                        position[0], position[1], position[2], pOrientationQuat[0],
-                        pOrientationQuat[1], pOrientationQuat[2], pOrientationQuat[3],
-                        pViewMatrix[0], pViewMatrix[1], pViewMatrix[2], pViewMatrix[3],
-                        pViewMatrix[4], pViewMatrix[5], pViewMatrix[6], pViewMatrix[7],
-                        pViewMatrix[8], pViewMatrix[9], pViewMatrix[10], pViewMatrix[11],
-                        pViewMatrix[12], pViewMatrix[13], pViewMatrix[14], pViewMatrix[15],
-                        pProjectionMatrix[0], pProjectionMatrix[1], pProjectionMatrix[2],
-                        pProjectionMatrix[3], pProjectionMatrix[4], pProjectionMatrix[5],
-                        pProjectionMatrix[6], pProjectionMatrix[7], pProjectionMatrix[8],
-                        pProjectionMatrix[9], pProjectionMatrix[10], pProjectionMatrix[11],
-                        pProjectionMatrix[12], pProjectionMatrix[13], pProjectionMatrix[14],
-                        pProjectionMatrix[15]];
-
+    // Get the camera frame in base 64.
+    NSString* base64ImageString = @"";
+    if (USE_CAMERA_FRAME) {
+      base64ImageString = [self getBase64ImageFromPixelBuffer:
+                           frame.capturedImage];
+      if (!base64ImageString) {
+        base64ImageString = @"";
+      }
+      else {
+        base64ImageString = [NSString stringWithFormat:
+                             @"data:image/jpg;base64, %@", base64ImageString];
+      }
+    }
+  
+    // Create a NSDictionary that will be parsed as a json and then passed to the JS side
+    NSDictionary* jsonDictionary = @{
+        @"position":@[FBOX(position[0]), FBOX(position[1]), FBOX(position[2])],
```

Comment:
I would recommend putting all of these "FBOX" lines at the continuation indent for better legibility instead of having them all aligned at different amounts based on the name of the property.

Reply:
Will try, XCode is not great for formatting stuff AFAIK.